core/variant/callable.cpp

```diff
@@ -124,9 +124,21 @@ Callable Callable::unbind(int p_argcount) const {
 bool Callable::is_valid() const {
 	if (is_custom()) {
 		return get_custom()->is_valid();
-	} else {
-		return get_object() && get_object()->has_method(get_method());
 	}
+
+	Object *target = get_object();
+	if (target) {
+		if (target->has_method(get_method())) {
+			return true;
+		}
+		// Script.has_method() and Object::has_method() have different semantics.
+		// Checking if a script (i.e., a class) has some method has to be done explicitly.
+		Script *script = Object::cast_to<Script>(target);
+		if (script && script->has_method(get_method())) {
+			return true;
+		}
```

Comment:
Perhaps it makes sense to check this in `Object::has_method()`, not in `Callable::is_valid()`? See updated #82767? I added `Script::has_static_method()` for better correctness (since non-static methods should not be called without an instance), but it will work with `Script::has_method()` as well.

Reply:
Your revised PR is indeed better than this one.